crates/net/eth-wire/src/ethstream.rs

```diff
@@ -73,7 +73,10 @@ where
 
         let msg = match ProtocolMessage::decode(&mut their_msg.as_ref()) {
             Ok(m) => m,
-            Err(err) => return Err(err.into()),
+            Err(err) => {
+                tracing::warn!("rlp decode error in handshake: msg={their_msg:x}");
```

Comment:
nit: there are multiple handshakes, so specifying `eth` would make it more clear where the decoding failed
```suggestion
                tracing::warn!("rlp decode error in eth handshake: msg={their_msg:x}");
```

Reply:
Done!